### PR TITLE
2701 rewards

### PR DESCRIPTION
**Pull request by Fesmofet**

### CODE REVIEW

File: controllers/validators/campaignsValidator.js [hunk 1]
```diff
@@ -143,3 +143,10 @@ exports.historySchema = Joi.object().keys({
   caseStatus: Joi.string().valid('open', 'close', 'all').default('all'),
   rewards: Joi.array().items(Joi.string().valid(...reservationStatuses)).default(reservationStatuses),
 });
+
+exports.validateUserRewardsSchema = Joi.object().keys({
+  skip: Joi.number().default(0),
+  limit: Joi.number().default(10),
+  simplified: Joi.boolean().default(false),
```
Comment (olegtsigulov) [inline controllers/validators/campaignsValidator.js:150]: You use simplified? i use this field for simplify response for map loading

File: swagger/swagger.json [hunk 2]
```diff
@@ -881,6 +881,69 @@
...
+            "name": "limit",
+            "in": "query",
+            "type": "number",
+            "description": "default - 10 "
+          },
+          {
+            "name": "simplified",
+            "in": "query",
+            "type": "boolean",
+            "description": "default - false "
```
Comment (olegtsigulov) [inline swagger/swagger.json:916]: ????????

File: test/controllers/campaignsController/test.js [hunk 3]
```diff
@@ -2428,3 +2425,56 @@ describe('blackList', async () => {
...
+      followers_count: 2,
+    });
+  });
+
+  it('route /campaigns-api/rewards/:userName should return 200', async () => {
+    const res = await chai.request(app).get(`/campaigns-api/rewards/${user.name}`);
+    res.should.have.status(200);
+  });
+
+  it('route /campaigns-api/rewards/:userName with wrong route should return 404', async () => {
```
Comment (olegtsigulov) [inline test/controllers/campaignsController/test.js:2448]: remove route /campaigns-api/rewards/:userName

File: test/controllers/campaignsController/test.js [hunk 4]
```diff
@@ -2428,3 +2425,56 @@ describe('blackList', async () => {
...
+    await SubscriptionFactory.Create({ follower: user.name, following: campaign.guideName });
+    const { body: { campaigns: [returnedCampaign] } } = await chai.request(app).get(`/campaigns-api/rewards/${user.name}`);
+    expect(returnedCampaign).to.be.exist;
+  });
+
+  it('if user follows campaign guide he should see campaign in feed', async () => {
+    await SubscriptionFactory.Create({ follower: user.name, following: campaign.guideName });
+    const { body: { campaigns: [returnedCampaign] } } = await chai.request(app).get(`/campaigns-api/rewards/${user.name}`);
+    expect(returnedCampaign).to.be.exist;
+  });
```
Comment (olegtsigulov) [inline test/controllers/campaignsController/test.js:2474]: double test?

File: utilities/operations/campaigns/getUserRewards.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
+const _ = require('lodash');
+const { userModel, Subscriptions } = require('models');
+const { campaignHelper } = require('utilities/helpers');
+
+module.exports = async (req, params) => {
+  const name = req.params.userName.toLocaleLowerCase();
```
Comment (olegtsigulov) [inline utilities/operations/campaigns/getUserRewards.js:6]: why you use it? valid username dont have upper case symbols

File: controllers/campaignsController.js [hunk 6]
```diff
@@ -210,6 +210,22 @@ const getCampaignsStatistic = async (req, res) => {
...
 
+const userRewards = async (req, res) => {
+  const { params, validationError } = validators
+    .validate(req.query, validators.campaigns.validateUserRewardsSchema);
+  if (validationError) return renderError(res, validationError);
+
+  const {
+    // eslint-disable-next-line camelcase
+    campaigns, campaigns_types, hasMore, sponsors, radius, error,
+  } = await getUserRewards(req, params);
```
Comment (olegtsigulov) [inline controllers/campaignsController.js:221]: no need to pass a request to a method

File: utilities/operations/campaigns/getUserRewards.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  campaigns = campaignHelper.eligibleCampaignsFilter(campaigns, name);
+
+  const resultArr = [...new Set(_.concat(
+    _.filter(campaigns, (el) => _.includes(users, el.guideName)),
+    _.filter(campaigns, (el) => (
+      _.includes(user.objects_follow, el.requiredObject) && name !== el.guideName)),
+  ))];
+  const {
+    skip, limit, simplified, sort,
+  } = params;
```
Comment (olegtsigulov) [inline utilities/operations/campaigns/getUserRewards.js:27]: you can destructurise all params here - module.exports = async (req, params) => {

File: utilities/operations/campaigns/getUserRewards.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+      campaigns: [], campaigns_types: [], hasMore: false, sponsors: [],
+    };
+  }
+  campaigns = campaignHelper.eligibleCampaignsFilter(campaigns, name);
+
+  const resultArr = [...new Set(_.concat(
+    _.filter(campaigns, (el) => _.includes(users, el.guideName)),
+    _.filter(campaigns, (el) => (
+      _.includes(user.objects_follow, el.requiredObject) && name !== el.guideName)),
+  ))];
```
Comment (olegtsigulov) [inline utilities/operations/campaigns/getUserRewards.js:24]: use just one filter with or conditions, why you check this? - && name !== el.guideName

Comment (Fesmofet) [reply utilities/operations/campaigns/getUserRewards.js:24]: fixed

File: utilities/operations/campaigns/getUserRewards.js (hunk 7 above)
Comment (Fesmofet) [reply utilities/operations/campaigns/getUserRewards.js:27]: fixed

File: controllers/campaignsController.js (hunk 6 above)
Comment (Fesmofet) [reply controllers/campaignsController.js:221]: fixed

File: utilities/operations/campaigns/getUserRewards.js (hunk 5 above)
Comment (Fesmofet) [reply utilities/operations/campaigns/getUserRewards.js:6]: fixed

File: test/controllers/campaignsController/test.js (hunk 4 above)
Comment (Fesmofet) [reply test/controllers/campaignsController/test.js:2474]: fixed

File: test/controllers/campaignsController/test.js (hunk 3 above)
Comment (Fesmofet) [reply test/controllers/campaignsController/test.js:2448]: fixed

File: swagger/swagger.json (hunk 2 above)
Comment (Fesmofet) [reply swagger/swagger.json:916]: fixed

File: controllers/validators/campaignsValidator.js (hunk 1 above)
Comment (Fesmofet) [reply controllers/validators/campaignsValidator.js:150]: removed

**Pull request merged**